server/app/mutations/domain_authorizations/authorize.rb

```diff
@@ -23,11 +23,16 @@ def validate
       unless grid.grid_secrets.find_by(name: LE_PRIVATE_KEY)
         add_error(:le_registration, :missing, "Let's Encrypt registration missing")
       end
-      if self.authorization_type == 'tls-sni-01'
-        add_error(:linked_service, :missing, "Service link needs to be given for tls-sni-01 authorization type") unless self.linked_service
+      if self.authorization_type == 'tls-sni-01' && self.linked_service.nil?
+        add_error(:linked_service, :missing, "Service link needs to be given for tls-sni-01 authorization type")
       end
       if self.linked_service
-        add_error(:linked_service, :not_found, "Linked service needs to point to existing service") unless @linked_service = resolve_service(self.grid, linked_service)
+        @lb_service = resolve_service(self.grid, linked_service)
```

Comment:
Originally, I tried to avoid using `lb_*` as variable names :), technically one can map the tls-sni cert into any service. Yes, in 99.99% of the cases it will be a LB

Reply:
Ok, the reason for variable name change was that it did override `linked_service` from mutation (that is a string).